src/shared/dissect-image.h

```diff
@@ -74,10 +74,8 @@ typedef enum DissectImageFlags {
                                                  DISSECT_IMAGE_MOUNT_READ_ONLY,
         DISSECT_IMAGE_GROWFS                   = 1 << 18, /* Grow file systems in partitions marked for that to the size of the partitions after mount */
         DISSECT_IMAGE_MOUNT_IDMAPPED           = 1 << 19, /* Mount mounts with kernel 5.12-style userns ID mapping, if file system type doesn't support uid=/gid= */
-        DISSECT_IMAGE_MANAGE_PARTITION_DEVICES = 1 << 20, /* Manage partition devices, e.g. probe each partition in more detail */
-        DISSECT_IMAGE_OPEN_PARTITION_DEVICES   = 1 << 21, /* Open dissected partitions and decrypted partitions */
-        DISSECT_IMAGE_BLOCK_DEVICE             = DISSECT_IMAGE_MANAGE_PARTITION_DEVICES |
-                                                 DISSECT_IMAGE_OPEN_PARTITION_DEVICES,
+        DISSECT_IMAGE_ADD_PARTITION_DEVICES    = 1 << 20, /* Create partition devices via BLKPG_ADD_PARTITION */
+        DISSECT_IMAGE_PIN_PARTITION_DEVICES    = 1 << 21, /* Open dissected partitions and decrypted partitions and pin them by fd */
```

Comment:
Does it make sense to have this as a separate flag? At least in this PR, you either specify none of these two flags or both of them, so why have two flags instead of one?

Reply:
well, the previous version of the code added a combined enum that merged the two flags under one name. I dropped that mostly because I couldn't find a clear name for it htat was better than the separate flags. 

The two operations of the two flags are really different I think. While they usually are used in combination I would still keep them seperate because they are so different.

I mean, I could even see a real-life usecase for this. For example, I think we probably should add "systemctl-dissect --attach" which validates a DDI and synchronously attaches a la "losetup -P", but doesn't actually mount it. Could be useful of you then want to mount it via mount units (potentially even dynamic automount units via systemd-mount) or via udisks or so. In that case we want the ADD but not the PIN.